crates/driver/src/infra/solver/dto/solution.rs

```diff
@@ -221,7 +227,11 @@ pub struct Solution {
     #[serde_as(as = "HashMap<_, serialize::U256>")]
     prices: HashMap<eth::H160, eth::U256>,
     trades: Vec<Trade>,
+    #[serde(default)]
+    pre_interactions: Vec<Interaction>,
```

Comment:
I think this is the wrong type. It doesn't make sense for solvers to specify Liquidity interactions as pre interactions. It also doesn't make sense for those to be internalizable nor specify input/output amounts. I think they should simply be 

```
struct {
  target: eth::H160,
  value: eth::U256,
  call_data: Vec<u8>,
}
```

Reply:
Switched to `InteractionData`